server/src/main/java/org/elasticsearch/action/admin/indices/dangling/NodeFindDanglingIndexResponse.java

```diff
@@ -0,0 +1,62 @@
+/*
+ * Licensed to Elasticsearch under one or more contributor
+ * license agreements. See the NOTICE file distributed with
+ * this work for additional information regarding copyright
+ * ownership. Elasticsearch licenses this file to you under
+ * the Apache License, Version 2.0 (the "License"); you may
+ * not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.elasticsearch.action.admin.indices.dangling;
+
+import org.elasticsearch.action.support.nodes.BaseNodeResponse;
+import org.elasticsearch.cluster.metadata.IndexMetaData;
+import org.elasticsearch.cluster.node.DiscoveryNode;
+import org.elasticsearch.common.io.stream.StreamInput;
+import org.elasticsearch.common.io.stream.StreamOutput;
+
+import java.io.IOException;
+import java.util.List;
+
+/**
+ * Used when querying every node in the cluster for a specific dangling index.
+ */
+public class NodeFindDanglingIndexResponse extends BaseNodeResponse {
+    /**
+     * A node could report several dangling indices. This class will contain them all.
+     * A single node could even multiple different index versions for the same index
+     * UUID if the situation is really crazy, though perhaps this is more likely
+     * when collating responses from different nodes.
+     */
+    private final List<IndexMetaData> danglingIndexMetaData;
```

Comment:
Can we avoid sending the full index metadata here? This object is potentially very big, and we are only interested in two small fields from that object.

Reply:
Note that we're only sending the metadata for a single index here. Although it might be quite big it shouldn't be unmanageably so.

There's a couple of options for trimming it down further: we could distinguish the `find` action needed for deletion from the one needed for import, or else we could share just the metadata version here and have the coordinating node pick one node and tells it to send its metadata to the master. Either way we have to add another transport action, and I don't think the savings are really worth it.

Reply:
ok, following your suggestion here